### ISSUE TITLE
Weird context sensitive error in FStar's parse

### ISSUE THREAD

**Issue by kyoDralliam** [fsprojects/FsLexYacc#68]
### Description

I am working on the fsyacc generated FStar parser ([parse.fsy](https://github.com/FStarLang/FStar/blob/kyod_parser/src/parser/parse.fsy)) and am experiencing some troublesome and hard to debug behaviours. Some productions seems to fail to parse if they are not parsed after some unrelated production. As an example the following does not parse :

````
module DoesNotParse

let x = ()
````
(as tokens : MODULE ; NAME "DoesNotParse" ; LET false ; IDENT "x" ; EQUALS ; LPAREN_RPAREN ; EOF )

whereas adding an unrelated declaration beforehand makes the parsing go through :

````
module DoesParse

val y : int
let x = ()
````
(as tokens : MODULE ; NAME "DoesParse" ; VAL ; IDENT "y" ; COLON ; IDENT "int" ; LET false ; IDENT "x" ; EQUALS ; LPAREN_RPAREN ; EOF)

### Repro steps

1. Build fsharp's FStar from kyod_parser branch (from top directory execute `make -C src`)

2. Execute the two previous files with fstar 
    `fstar.exe --explicit_deps doesNotParse.fst` 
    `fstar.exe --explicit_deps doesParse.fst`

### Expected behavior

Any `let` declaration should be able to parse even just after the `module` declaration (using productions `input_fragment ; decl ; decl2`). It works fine with both the ocamlyacc and menhir parser generators from which the fsyacc parser is generated (the transformation from ocamlyacc format to fsyacc format is just a line of sed).

### Actual behavior

Such a `let` declaration does not parse unless there is a `val` declaration beforehand (which is also produced by `decl2`). This misbehaviour has also been witnessed with some other tokens previously (but which seems more difficult to debug).

Tested with nuget packages 6.1.0 and 7.02 (for both FsLexYacc and the runtime). I will try to minimize the parser to find where the error is comming from. I also would like to use the debug facilites in fsyacc (I saw in the code that there is some Flag.debug which can be enabled if __DEBUG is defined at compile time) but I didn't manage to build Fsyacc from sources... Is there some way to have this enabled with a nuget package ?

**Comment by kyoDralliam** [comment]
I just found out where the error was comming from : I was using erroneous queries on the position of tokens. I guess I have been misled by the different behaviour of ocamlyacc/menhir which seems to return the range information for the current production and silently drop any incorrect range request.

Also I just realised that FStar was dropping the raised exception : 
````error : System.NullReferenceException: Object reference not set to an instance of an object````
which comes from a call to `parseState.InputRange(n)` (`parseState` being a `Microsoft.FSharp.Text.Parsing.IParseState`).
